app/models/concerns/purgable.rb

```diff
@@ -0,0 +1,78 @@
+# Copied from the deprecated fastly-rails gem
+# https://github.com/fastly/fastly-rails/blob/master/lib/fastly-rails/active_record/surrogate_key.rb
+#
+# This concern handles purge and purge_all calls to purge the edge cache (Fastly)
+module Purgable
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def purge_all
+      return unless Rails.env.production?
+
+      service.purge_by_key(table_key)
+    end
+
+    def soft_purge_all
+      return unless Rails.env.production?
+
+      service.purge_by_key(table_key, true)
+    end
+
+    def table_key
+      table_name
+    end
+
+    def fastly_service_identifier
+      return unless Rails.env.production?
+
+      service.service_id
+    end
+  end
+
+  # Instance methods
+  def record_key
+    "#{table_key}/#{id}"
+  end
+
+  def table_key
+    self.class.table_key
+  end
+
+  def purge
+    return unless Rails.env.production?
+
+    service.purge_by_key(record_key)
+  end
+
+  def soft_purge
+    return unless Rails.env.production?
+
+    service.purge_by_key(record_key, true)
+  end
+
+  def purge_all
+    self.class.purge_all
+  end
+
+  def soft_purge_all
+    self.class.soft_purge_all
+  end
+
+  def fastly_service_identifier
+    self.class.fastly_service_identifier
+  end
+
+  private
+
+  def fastly
+    return unless Rails.env.production?
```

Comment:
Technically this line isn't needed since `service` has the same guard and it's the only thing using it. If we add `purging` by url/path to the models, it'll need `fastly` by itself. I'd rather have the guard and not need it instead of forgetting to add it later on where we end up busting prod cache from development. That _especially_ comes into play when we bust by url.